### ISSUE TITLE
Linux dependencies

### ISSUE THREAD

**Issue by bb30994** [FoldingAtHome/fah-issues#1273]
This is an increasingly important issue.

> unfortunately 19.04 has done away with or changed python in such a way that I haven't been able to install FAHControl but I've got it on other systems so it's not a complete disaster

>Perhaps they've gone to python 3?

>Yes, I believe it's 3.7 after using 3.6 on 18.10 but I had no problem installing FAHControl on 18.10 so I'm not sure what is going on ... I've read that if you upgrade there's no problem but a new install will have problems ( https://www.phoronix.com/scan.php?page=news_item&px=Ubuntu-19.04-Development-Opens ) 
> 19.04 is still not officially released so someone may sort the problem or jcoffland may incorporate all dependencies in the software 

>I think FAH still uses python 2 ...

> yes ... that is what installing FAHControl on 19.04 is complaining about but for some reason I've been unable to install python 2 it keeps complaining about unsatisfiable dependencies

**Comment by beer118** [comment]
I manage to install FAH on Debian Buster by adding repoes from Debian oldStable. Those repose are being archive as we speak.  So as soon as Buster becomes stable then new users will not be able to install Folding@home on Debian stable anymore since the dependencies has been removed completely.

Since there are few distro's that is a fork from Debian (such as Ubuntu) then I would expect more and more people in the same boat as me.

**Comment by ProDigit** [comment]
Python 2.7 is being discontinued. Ubuntu 18 still supports it, but fahcontrol needs either python 3 or something other compatible.
Ubuntu 19 is out, and doesn't support python 2 anymore.
Fahcontrol is the main interface of fah.
I suggest prioritizing this one!

**Comment by ssbarnea** [comment]
The rpm is useless for CentOS 8, as it requires python executable which is not installed by any package. Even if python27 is also installed, you will not be able to install the rpm due to broken deps.

**Comment by amirkaveh** [comment]
FAHControl has the python-gnome2 package as its dependency while it is no longer supported in Ubuntu 19. Please resolve this issue. 

P.S. The issue can be resolved temporarily by removing this dependency from inside of .deb installer package, but it is complicated for end-users.

**Comment by orjanv** [comment]
I can confirm that editing the .deb as @amirkaveh writes works fine on Ubuntu 19.10 x86_64, but then I had some issues with a missing gtk module. I managed to solve this by installing the python-gtk2 module:

sudo apt-get install python-gtk2

Now I can start FAHControl, but no packages are retrieved. I started FAHClient in the shell and it seemed to work, at least the FAHControl connected to the client running, but a different error arose:

WARNING:WU00:FS00:Failed to get assignment from '18.218.241.186:80': No WUs available for this configuration
ERROR:WU00:FS00:Exception: Could not get an assignment

Any tips?

**Comment by amirkaveh** [comment]
try these:
sudo apt install python-gnome2 
sudo apt install --fix-broken

**Comment by orjanv** [comment]
Left it working for a while, now it seems to work:

09:26:00:WU00:FS00:Requesting new work unit for slot 00: READY cpu:1 from 128.252.203.9
09:26:00:WU00:FS00:Connecting to 128.252.203.9:8080
09:26:02:WU00:FS00:Downloading 8.14MiB
09:26:08:WU00:FS00:Download 79.87%
09:26:10:WU00:FS00:Download complete
09:26:10:WU00:FS00:Received Unit: id:00 state:DOWNLOAD error:NO_ERROR project:13827 run:446 clone:3 gen:3 core:0xa7 unit:0x0000000480fccb095e6d0c97096c8034
09:26:10:WU00:FS00:Downloading core from http://cores.foldingathome.org/v7/lin/64bit/avx/Core_a7.fah
09:26:10:WU00:FS00:Connecting to cores.foldingathome.org:80
09:26:11:WU00:FS00:FahCore a7: Downloading 8.91MiB
09:26:17:WU00:FS00:FahCore a7: 5.61%
09:26:23:WU00:FS00:FahCore a7: 7.02%

**Comment by boulabiar** [comment]
I have problems installing fahclient on ubuntu 19.10.
Tried everything up and still getting errors.
Please can someone share how to install the package or update them so they work?
I don't need a graphical interface (and many dependencies), a text interface with minimal information is sufficient.

**Comment by ggiovinazzo** [comment]
I found a useful script in order to modify deb dependencies.
And now i have a f@home control Gui working :dancer: 

https://gist.github.com/ggiovinazzo/ed7bc85b3c6797f318d228298cc93ea2

**Comment by ptrm** [comment]
> I found a useful script in order to modify deb dependencies.
> And now i have a f@home control Gui working 

Confirming this script works, replacing `Dependencies: python-gnome2` with `Dependencies: python-gtk2` also seems safe enough dependency-wise, as the control panel appears to be working.

@ggiovinazzo btw thanks, I read your comment 5 minutes after you posted it while browsing google for solutions :D

**Comment by boulabiar** [comment]
I tested this for fahcontrol deb and I can open a GUI.
but what about for fahclient ? It can't also be installed. How to fix it?

**Comment by ptrm** [comment]
> I tested this for fahcontrol deb and I can open a GUI (can't connect to anything).
> For fahclient, it wasn't able to install the first time because I haven't checked the option to launch it at startup (it fails instead). If checked, I can run it after install. I also had to open the port 8080.

@boulabiar try `dpkg-reconfigure <package-name>` from console, it should reopen the config windows and let you set the options again

**Comment by ptrm** [comment]
> Confirming this script works, replacing `Dependencies: python-gnome2` with `Dependencies: python-gtk2` (rest of the line unchanged) also seems safe enough dependency-wise, as the control panel appears to be working.

To be more specific, I did the above on Ubuntu 19.10 as well

**Comment by Oumpa31** [comment]
I am having the same issue fahcontrol wont install at all. How can i edit the dependencies. I'm new to editing these things but i'd like to put my computer to good use when I'm not using it. 
I'm not really sure how to use that script and how to put it together so i can fix fahcontrol_7.5.1-1_all.deb.
I have Ubuntu 19.10

**Comment by ptrm** [comment]
@Oumpa31:
> I'm not really sure how to use that script and how to put it together so i can fix fahcontrol_7.5.1-1_all.deb.

Here's a cast, it should help: https://asciinema.org/a/vNrGMcHdhFPGpyU2wDD5rW5LW

In short, you run ```
$ chmod +x deb-strip-dependencies.sh
$ ./deb-strip-dependencies.sh fahcontrol_7.5.1-1_all.deb
```

A default editor for you system will appear, you edit the line `Description: ...` to contain `python-gtk2` instead of `python-gnome2` (in the cast it is edited to have alternative dependencies).

After saving and exiting, you run
```
dpkg -i fahcontrol_7.5.1-1_all.modfied.deb
```

which sould no longer require the missing python-gnome2 and install smoothly.

If dpkg shows error about any other package, you should be able to install it via `sudo apt-get install package1 package2 ...`.

**Comment by mandree** [comment]
FTR, Fedora 31 Linux also defaults to Python 3, `python2 /usr/bin/FAHControl` works for now (assuming the requisites have been installed), but Fedora are also phasing out Python 2.x.   

Perhaps py-future and futurize can assist with the migration to Python 3.x.

**Comment by Dominique-M** [comment]
File-roller opens deb files read-only. I had to rename .deb in .ar,, open with file-roller, go to the control file, extract, edit -gnome2 → -gtk2, put the control file back, close (update yes). Rename again .ar in .deb and it works (Ubuntu 19.10).

**Comment by MarkP2015** [comment]
> @Oumpa31:
> 
> > I'm not really sure how to use that script and how to put it together so i can fix fahcontrol_7.5.1-1_all.deb.
> 
> Here's a cast, it should help: https://asciinema.org/a/vNrGMcHdhFPGpyU2wDD5rW5LW
> 
> In short, you run
> 
> ```
> $ chmod +x deb-strip-dependencies.sh
> $ ./deb-strip-dependencies.sh fahcontrol_7.5.1-1_all.deb
> ```
> 
> A default editor for you system will appear, you edit the line `Depends: ...` to contain `python-gtk2` instead of `python-gnome2` (in the cast it is edited to have alternative dependencies).
> 
> After saving and exiting, you run
> 
> ```
> $ sudo dpkg -i fahcontrol_7.5.1-1_all.modfied.deb
> ```
> 
> which sould no longer require the missing python-gnome2 and install smoothly.
> 
> If dpkg shows error about any other package, you should be able to install it via `sudo apt-get install package1 package2 ...`.

I cannot make the modified file "save and exit". I have no idea what app that text appears in. My file has, as the last line, the @ in color blue. I hope this helps.

> @Oumpa31:
> 
> > I'm not really sure how to use that script and how to put it together so i can fix fahcontrol_7.5.1-1_all.deb.
> 
> Here's a cast, it should help: https://asciinema.org/a/vNrGMcHdhFPGpyU2wDD5rW5LW
> 
> In short, you run
> 
> ```
> $ chmod +x deb-strip-dependencies.sh
> $ ./deb-strip-dependencies.sh fahcontrol_7.5.1-1_all.deb
> ```
> 
> A default editor for you system will appear, you edit the line `Depends: ...` to contain `python-gtk2` instead of `python-gnome2` (in the cast it is edited to have alternative dependencies).
> 
> After saving and exiting, you run
> 
> ```
> $ sudo dpkg -i fahcontrol_7.5.1-1_all.modfied.deb
> ```
> 
> which sould no longer require the missing python-gnome2 and install smoothly.
> 
> If dpkg shows error about any other package, you should be able to install it via `sudo apt-get install package1 package2 ...`.

Not knowing much about this, after changing to python-gtk2 and finding a way to save the program, by (shift+z+z, that is: shift z z) fahcontrol_7.5.1-1_all.deb, which was then renamed with the "modfied" word, I renamed the unmodified version by adding .original in the file name. Next sudo dpkg -i fahcontrol_7.5.1-1_all.deb - again the terminal said failed dependencies. And I cannot connect to the F@H via the webgui. Is using the "modfied" critical to this???

**Comment by ptrm** [comment]
@MarkP2015 
> I cannot make the modified file "save and exit". I have no idea what app that text appears in.
Try
```
EDITOR=/usr/bin/gedit ./deb-strip-dependencies.sh fahcontrol_7.5.1-1_all.deb 
```

> Next sudo dpkg -i fahcontrol_7.5.1-1_all.deb - again the terminal said failed dependencies. And I cannot connect to the F@H via the webgui. Is using the "modfied" critical to this???

No, the GUI package is for showing windowed version in addition to the webpage one. The only *crucial* package is the fahclient one, which should run both the work units and the web ui.

If you missed the settings during installation, run
```
sudo dpkg-reconfigure fah-client
```
It should ask you whether to run fahclient automatically, to which you can say yes for convenience.

There's also a known bug for the web gui on Chrome: https://foldingforum.org/viewtopic.php?f=61&t=32463 (Look for "7. How to fix Web control error ?").
Follow the instructions there, or just try visiting http://localhost:7396/ from Firefox (browser like Chromium and Vivaldi *also* use webkit engine, so they will behave the same as Chrome).

**Comment by MarkP2015** [comment]
Thank you for a swift response. Trying to reconfigure: 
mark@Lexington:~$ sudo dpkg-reconfigure fah-client
[sudo] password for mark: 
dpkg-query: package 'fah-client' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files.
/usr/sbin/dpkg-reconfigure: fah-client is not installed
As for known bug on Chrome, I use Firefox.

**Comment by ptrm** [comment]
@MarkP2015:
> /usr/sbin/dpkg-reconfigure: fah-client is not installed
> As for known bug on Chrome, I use Firefox.

I made a typo in the first version of my previous reply, it should be `fahclient`, not `fah-client`.

As for the web ui problems, that might be off topic in this issue thread, though without any screenshot or error message it is hard to help anyway. Why don't you try posting on [foldingforum.org](https://foldingforum.org), giving the details like screenshots, errors etc.? Try also viewing the rest of the FAQ I linked to above, it might be helpful in other problems, too.

**Comment by Evo221** [comment]
I think the fastest work around to get fahcontrol installed is:
'''
 sudo dpkg -i --ignore-depends=python-gnome2 fahcontrol_7.5.1-1_all.deb
'''

**Comment by Evo221** [comment]
Hi,

quick fix:

```
ar x fahcontrol_7.5.1-1_all.deb
tar xf control.tar.gz
sed -i 's/7.5.1-1/7.5.1-2/g' control
sed -i 's/python-gnome2/python-gtk2/g' control
rm control.tar.gz
tar cf control.tar.gz control md5sums
ar rcs fahcontrol_7.5.1-2_all.deb debian-binary control.tar.gz data.tar.xz
sudo dpkg -i fahcontrol_7.5.1-2_all.deb
```

**Comment by MarkP2015** [comment]
mark@Lexington:~/Downloads/fah7511$ tar xf control.tar.gz

tar: control.tar.gz: Cannot open: No such file or directory

tar: Error is not recoverable: exiting now


mark@Lexington:~/Downloads/fah7511$ ls

control.tar.xz
fahclient_7.5.1.modfied_amd64.deb
data.tar.xz
fahcontrol_7.5.1-1_all.deb
debian-binary
fahcontrol_7.5.1-1_all.deb.original
deb-strip-dependencies.sh
fahviewer_7.5.1_amd64.deb

________________________________

**Comment by Evo221** [comment]
@MarkP2015 it seems you missed the first step of unupacking the deb file:
`ar x fahcontrol_7.5.1-1_all.deb`

Also, althought what I posted will work I forgot to compress when taring control and md5sums. So that line would be better as:
```
tar czf control.tar.gz control md5sums
```

**Comment by MarkP2015** [comment]
Please understand that I am not a coder. I can follow instructions, usually with copy and paste. Beyond that I am all-in ignorance.
I ran Synaptic to see if any f@h was installed. It finds none. I shift-deleted the existing objects in my /fah7511 folder. I downloaded fresh .deb at Folding@Home.org I ran the instructions as you give them. See terminal output, below:

mark@Lexington:~/Downloads/fah7511$ ls
fahclient_7.5.1_amd64.deb   fahviewer_7.5.1_amd64.deb
fahcontrol_7.5.1-1_all.deb
mark@Lexington:~/Downloads/fah7511$ tar czf control.tar.gz control md5sums
tar: control: Cannot stat: No such file or directory
tar: md5sums: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
mark@Lexington:~/Downloads/fah7511$ ar x fahcontrol_7.5.1-1_all.deb
mark@Lexington:~/Downloads/fah7511$ ls
control.tar.gz  data.tar.xz  debian-binary  fahclient_7.5.1_amd64.deb  fahcontrol_7.5.1-1_all.deb  fahviewer_7.5.1_amd64.deb
mark@Lexington:~/Downloads/fah7511$ tar czf control.tar.gz control md5sums
tar: control: Cannot stat: No such file or directory
tar: md5sums: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
mark@Lexington:~/Downloads/fah7511$

**Comment by Evo221** [comment]
@MarkP2015 you need to issue the commands in the correct order. When posting output from your terminal please put it in a code block using three back ticks: ```

**Comment by MarkP2015** [comment]
I have re-issued them, following this order:
`ar x fahcontrol_7.5.1-1_all.deb
tar xf control.tar.gz
sed -i 's/7.5.1-1/7.5.1-2/g' control
sed -i 's/python-gnome2/python-gtk2/g' control
rm control.tar.gz
tar cf control.tar.gz control md5sums
ar rcs fahcontrol_7.5.1-2_all.deb debian-binary control.tar.gz data.tar.xz
sudo dpkg -i fahcontrol_7.5.1-2_all.deb`

and Folding Control shows and at least starts on my desktop. It reads: Connecting in reverse orange background text. I'll give it a few hours to see if it can get a WU. Blessings, Evo221

**Comment by Evo221** [comment]
@MarkP2015 have you installed the fahclient package? That is what actually does the work. And if it is installed is it running? Eg on my machine:
```
% dpkg -l --no-pager fahclient
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  fahclient      7.5.1        amd64        Folding@home Console Client

% systemctl status --no-pager FAHClient
● FAHClient.service - LSB: Folding@home Client
   Loaded: loaded (/etc/init.d/FAHClient; generated)
   Active: active (running) since Tue 2020-03-31 11:05:03 JST; 1 day 22h ago
     Docs: man:systemd-sysv-generator(8)
    Tasks: 31 (limit: 4915)
   Memory: 253.1M
   CGroup: /system.slice/FAHClient.service
           ├─ 1607 /usr/bin/FAHClient /etc/fahclient/config.xml --run-as fahclient --pid-file=/var/run…
           ├─ 1609 /usr/bin/FAHClient --child --lifeline 1607 /etc/fahclient/config.xml --run-as fahcl…
           ├─24875 /usr/bin/FAHCoreWrapper /var/lib/fahclient/cores/cores.foldingathome.org/v7/lin/64b…
           └─24879 /var/lib/fahclient/cores/cores.foldingathome.org/v7/lin/64bit/avx/Core_a7.fah/FahCo…

Mar 31 11:05:02 xps systemd[1]: Starting LSB: Folding@home Client...
Mar 31 11:05:03 xps FAHClient[1597]: Starting fahclient ... OK
Mar 31 11:05:03 xps systemd[1]: Started LSB: Folding@home Client.
```

**Comment by MarkP2015** [comment]
@Evo221

If as you write ```sudo dpkg -l --no-pager fahclient```
installs the FAHControl, then from this terminal output, I guess it isn't installed. I tried the line you gave
```
mark@Lexington:~/Downloads/fah7511$ sudo dpkg -l --no-pager fahclient
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
un  fahclient      <none>       <none>       (no description available)
```


and

```
mark@Lexington:~/Downloads/fah7511$ % systemctl status --no-pager FAHClient
bash: fg: %: no such job
```

```
mark@Lexington:~/Downloads/fah7511$ ls
control
debian-binary
fahcontrol_7.5.1-2_all.deb
control.tar.gz
fahclient_7.5.1_amd64.deb
fahviewer_7.5.1_amd64.deb
data.tar.xz
fahcontrol_7.5.1-1_all.deb
md5sums
```

**Comment by Evo221** [comment]
@MarkP2015 that shows that fahclient isn't installed. to install it do:
```
cd ~/Downloads/fah7511
sudo dpkg -i fahclient_7.5.1_amd64.deb
```
It will ask if you want to run automatically at start up: you should probbaly say yes, otherwise you'll need to learn to start it manually.
You can also install the 3D visualisation if you desire:
```
sudo dpkg -i fahviewer_7.5.1_amd64.deb
```

**Comment by MarkP2015** [comment]
@Evo221
It does not run. The `https://client.foldingathome.org/ `never shows a connection. Below, is me, today, trying to follow your instructions.

```
mark@Lexington:~$ cd /home/mark/Downloads/fah7511
mark@Lexington:~/Downloads/fah7511$ ls
 control          fahclient_7.5.1_amd64.deb   'fah no connection.png'
 control.tar.gz   fahcontrol_7.5.1-1_all.deb   fahviewer_7.5.1_amd64.deb
 data.tar.xz      fahcontrol_7.5.1-2_all.deb   md5sums
 debian-binary    FaH-control.png

mark@Lexington:~/Downloads/fah7511$ ar x fahcontrol_7.5.1-1_all.deb
mark@Lexington:~/Downloads/fah7511$ tar xf control.tar.gz
mark@Lexington:~/Downloads/fah7511$ sed -i 's/7.5.1-1/7.5.1-2/g' control
mark@Lexington:~/Downloads/fah7511$ sed -i 's/python-gnome2/python-gtk2/g' control
mark@Lexington:~/Downloads/fah7511$ rm control.tar.gz
mark@Lexington:~/Downloads/fah7511$ tar cf control.tar.gz control md5sums
mark@Lexington:~/Downloads/fah7511$ ar rcs fahcontrol_7.5.1-2_all.deb debian-binary control.tar.gz data.tar.xz

mark@Lexington:~/Downloads/fah7511$ sudo dpkg -i fahcontrol_7.5.1-2_all.deb
[sudo] password for mark: 
(Reading database ... 281952 files and directories currently installed.)
Preparing to unpack fahcontrol_7.5.1-2_all.deb ...
Unpacking fahcontrol (7.5.1-2) over (7.5.1-2) ...
Setting up fahcontrol (7.5.1-2) ...
Processing triggers for desktop-file-utils (0.23-4) ...
Processing triggers for mime-support (3.62) ...

I then try: https://client.foldingathome.org/  -- fail

mark@Lexington:~/Downloads/fah7511$ ls
 control          fahclient_7.5.1_amd64.deb   'fah no connection.png'
 control.tar.gz   fahcontrol_7.5.1-1_all.deb   fahviewer_7.5.1_amd64.deb
 data.tar.xz      fahcontrol_7.5.1-2_all.deb   md5sums
 debian-binary    FaH-control.png

mark@Lexington:~/Downloads/fah7511$ sudo dpkg **--force-all -i f**ahclient_7.5.1_amd64.deb
(Reading database ... 281952 files and directories currently installed.)
Preparing to unpack fahclient_7.5.1_amd64.deb ...
invoke-rc.d: policy-rc.d denied execution of stop.
usermod: no changes
Unpacking fahclient (7.5.1) over (7.5.1) ...
Setting up fahclient (7.5.1) ...
**Failed to enable unit: Unit /run/systemd/generator.late/FAHClient.service is transient or generated.**
invoke-rc.d: policy-rc.d denied execution of start.

The Folding@home client is now installed

You can access the Web interface by going to:

  https://client.foldingathome.org/

in a browser on this computer.

For information about configuring the Folding@home client for 
remote access please see the comments in:

  /etc/fahclient/config.xml


Processing triggers for systemd (241-7~deb10u3) ...
Processing triggers for desktop-file-utils (0.23-4) ...
Processing triggers for mime-support (3.62) ...

Try: https://client.foldingathome.org/  --- Fail again

mark@Lexington:~/Downloads/fah7511$ 


```

**Comment by boulabiar** [comment]
I think someone should put the updated packages on the website instead of
the instructions on how to modify the old (current) ones.

Someone with the authorization to do it of course.

Le sam. 4 avr. 2020 à 00:01, Mark Preston <notifications@github.com> a
écrit :

**Comment by Evo221** [comment]
@MarkP2015 your problem seems to be:
```
invoke-rc.d: policy-rc.d denied execution of start.
```
What operating system are you using?

**Comment by MarkP2015** [comment]
@Evo221 
After much trial and tribulation, F@H is working. My 6 core cpu is being used by F@H on 5 cores and running them near 100% and not heating the cpu die/chip over an extended period of time. Thank you for your patience, goodwill and generosity with your time for me. Onward towards a cure!

**Comment by dmc5179** [comment]
@ssbarnea We have been able to install and run fahclient on RHEL 7/8, Centos 7/8, Fedora 31, Ubuntu 18.04, Amazon Linux 2. Are you referring to the FAHControl itself?

Ansible playbooks to install on the above OSs

https://github.com/ajacocks/fah

**Comment by ssbarnea** [comment]
@dmc5179 Sorry but I removed FAH from all machines I own and I will refuse to endorse such a tool that is not open source. While I am not an open-source purist and I do also use non-open-source software I do not think that this kind of software should ever be considered for not being open-source for two main reasons: **security** and **transparency**. While the main goals are good I cannot see any reasons to donate my money (especially electricity bill) to something that is I do not what it does. I have no practical way to discover that it does not harvest bitcoins for someone else today or tomorrow. It feels like donating money to a charity that does publish its accounts anywhere, where its board could as well spending money inappropriately.

I these would have being different I would have not mind helping with packaging and porting to py3, things that I do happen to have a lot of experience with.

**Comment by shorttack** [comment]
Join the discussion at [](https://foldingforum.org)

**Comment by boulabiar** [comment]
At what?

**Comment by PantherX** [comment]
@ssbarnea 
Please note that there are plans to make FAHControl open source. It is in the pipeline and needs a bit of time. If you have any concerns or questions, we would be happy to answer them in the Official F@H Forum here: https://foldingforum.org/index.php

**Comment by shorttack** [comment]
The current beta is located at [https://foldingathome.org/beta/](url)

**Comment by Oumpa31** [comment]
just installed this on Ubuntu 19.10 and it works flawlessly.  Thank you soooo much for posting this

> The current beta is located at [https://foldingathome.org/beta/](url)

**Comment by rpremuz** [comment]
While installing fahcontrol_7.6.21-1_all.deb on Ubuntu 20.04 64-bit it reports missing dependencies:
$ sudo gdebi fahcontrol_7.6.21-1_all.deb 
Reading package lists... Done
Building dependency tree        
Reading state information... Done
Reading state information... Done
This package is uninstallable
Dependency is not satisfiable: python-gtk2|python-gnome2

Neither python-gtk2 nor python-gnome2 are available on Ubuntu 20.04.

**Comment by PantherX** [comment]
Hiya @rpremuz 

Please note that work is underway here to move away from Python 2 to Python 3: https://github.com/FoldingAtHome/fah-control/pulls
